File: backend/src/main/scala/domains/problem/objects/response/ProblemSuggestion.scala
package domains.problem.objects.response

import domains.problem.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 题目搜索建议响应；只返回可展示的 slug 和标题。 */
final case class ProblemSuggestion(
  slug: ProblemSlug,
  title: ProblemTitle
)

/** ProblemSuggestion 的 JSON 编解码器。 */
object ProblemSuggestion:
  given Encoder[ProblemSuggestion] = deriveEncoder[ProblemSuggestion]
  given Decoder[ProblemSuggestion] = deriveDecoder[ProblemSuggestion]

File: frontend/src/objects/problem/response/ProblemSuggestion.ts
import type { ProblemSlug } from '@/objects/problem/ProblemSlug'
import type { ProblemTitle } from '@/objects/problem/ProblemTitle'

/** 题目搜索建议；只暴露用于选择的 slug 和标题。 */
export type ProblemSuggestion = {
  slug: ProblemSlug
  title: ProblemTitle
}
